nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -255,14 +257,9 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 throw new IllegalArgumentException(format("Statement Type is not specified, FlowFile %s", inputFlowFile));
             }
 
-            // throw an exception if the statement type is set to update and updateMethod or updateKeys is empty
-            if (UPDATE_TYPE.getValue().equalsIgnoreCase(statementType)) {
-                if (StringUtils.isEmpty(updateMethod)) {
-                    throw new IllegalArgumentException(format("Update Method is not specified, FlowFile %s", inputFlowFile));
-                }
-                if (StringUtils.isEmpty(updateKeys)) {
+            // throw an exception if the statement type is set to update and updateKeys is empty
+            if (UPDATE_TYPE.getValue().equalsIgnoreCase(statementType) && StringUtils.isEmpty(updateKeys)) {
```

Comment:
Checking `updateMethod` _here_ actually makes sense because it can come from a flowfile attribute and can be null/empty.
I'd leave this check as it was.

Reply:
I've added a test for this, but this is already checked on lines 355 to 367 (exception thrown on 366).

Reply:
Okay, that seems to cover it.